### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by tradersnow222**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,69 +1,186 @@
...
+
+### LP-Token farming
+Some protocols incentivize LP token holders with additional rewards beyond swap fees through an additional yield contract that holds the LP tokens and distributes the rewards proportionally to the participating LPs. With token plugins, this extra rewards for LP holders can continue to be opt-in without the need to deposit those LP tokens into a secondary contract. (See [1inch Fusion pods](https://etherscan.io/address/0x806d9073136c8A4A3fD21E0e708a9e17C87129e8#code))
+
+### Shadow staking
+If a protocol wanted to simply reward holders of their token, they could reward them similarly to the weighted voting power method, but instead of increasing voting power over time, the APR of holding the token can increase. Long-term holders will receive rewards and short-term holders/traders would not receive the same benefit.
+
+
```
Comment (galekseev) [inline README.md:57]: Another possible appliance is to reward user for lending or borrowing in lending protocols

File: README.md [hunk 2]
```diff
@@ -1,69 +1,186 @@
...
+For security purposes, plugins are designed with several fail-safes, including a maximum number of usable plugins, custom gas limits, a reentrancy guard, and native isolation from the main contract state. The maximum plugins and gas limit can be initialized as state variables using `MAX_PLUGINS_PER_ACCOUNT` and `PLUGIN_CALL_GAS_LIMIT`, respectively. For reentrancy prevention, `ReentrancyGuardExt` is included from OpenZeppelin’s library. Finally, for native isolation from the token contract, a single method with only three arguments (To, From, and Amount) is used. This simple architecture results in a dynamic (and risk-free!) enhancement of any ERC20 contract’s capabilities.
 
-## How does it work and what can I extend?
+### Integrating plugin support in your token implementation
+Technically, plugins are a collection of smart contracts that track changes in ERC20 token balances and perform supplementary accounting tasks for those balances. They are particularly useful when you need to track, for example, token shares without actually transferring tokens to a dedicated accounting contract.
```
Comment (galekseev) [inline README.md:81]: Fully duplicates line 26

File: README.md [hunk 3]
```diff
@@ -1,69 +1,186 @@
...
+
+### Integrating plugin support in your token implementation
+
+To integrate plugins in a smart contract, a "mothership" or parent contract must be used to manage all related pods. This includes adding, removing, and viewing pods, as well as connecting multiple pods. The parent contract calls the `updateBalance` function for each pod on every update of an account’s balance. The pod then executes its logic based on the updated balance information. An account must connect a plugin to utilize its logic.
+To integrate plugins in a smart contract, a "mothership" or parent contract must be used to manage all related pods. This includes adding, removing, and viewing pods, as well as connecting multiple pods. The parent contract calls the `updateBalance` function for each pod on every update of an account’s balance. The pod then executes its logic based on the updated balance information. An account must connect a plugin to utilize its logic.
```
Comment (galekseev) [inline README.md:138]: Duplicates previous line

File: README.md [hunk 4]
```diff
@@ -1,69 +1,186 @@
...
+To create your own plugin, it is necessary to inherit the `Plugin` contract and implement its abstract function `_updateBalances`.
+To create your own plugin, it is necessary to inherit the `Plugin` contract and implement its abstract function `_updateBalances`.
+
+1. **Inherit plugin**: `contract MyPlugin is Plugin { ... }`
+2. **Implement `_updateBalances` function** to process wallet balance changes.
+1. **Inherit plugin**: `contract MyPlugin is Plugin { ... }`
+2. **Implement `_updateBalances` function** to process wallet balance changes.
+
+## Generic Examples of a plugin-compatible ERC20 token and plugin
+## Generic Examples of a plugin-compatible ERC20 token and plugin
```
Comment (galekseev) [inline README.md:180]: Duplicates the previous line

File: README.md [hunk 5]
```diff
@@ -1,69 +1,186 @@
...
+
+Below is an example of a token implementing plugin support and a simple plugin that mints and burns its own token based on the parent’s token balance.
+Below is an example of a token implementing plugin support and a simple plugin that mints and burns its own token based on the parent’s token balance.
```
Comment (galekseev) [inline README.md:183]: Duplicates the previous line

File: README.md [hunk 6]
```diff
@@ -96,3 +213,31 @@ contract PluginExample is ERC20, Plugin {
...
+- [1inch Fusion Staking Farm](https://etherscan.io/address/0x1A87c0F9CCA2f0926A155640e8958a8A6B0260bE#code)
+
+### Other Helpful Links
+
+- [Plugin-enabled ERC20 Token contract (abstract)](https://github.com/1inch/token-plugins/blob/master/contracts/ERC20Plugins.sol)
+- [Plugin contract (abstract)](https://github.com/1inch/token-plugins/blob/master/contracts/Plugin.sol)
+- [Anton Bukov speech at ETHCC](https://youtu.be/Is-T5Q2E0A8?feature=shared)
+- [Kirill Kuznetcov speech at Nethermind Summit, Istanbul](https://youtu.be/BwehZHhR8Z4?feature=shared)
+
+### Deployed Examples
```
Comment (galekseev) [inline README.md:231]: Duplicates lines 217-229

**Review by galekseev — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -1,69 +1,186 @@
...
+Token plugins are smart contracts that extend the capabilities of ERC20 tokens and wrappers by adding custom accounting features to the original token. Inspired by the plugin concept widely used in the web 2.0 world, these plugins enable users to dynamically enhance the functionality of their tokens by subscribing to various plugins without the need to transfer tokens to a special smart contract.
+
+The major benefit, and a key difference from existing solutions, is that these do not require token transfers to a special smart contract, as is commonly seen in farming or delegating protocols. Another beneficial point is that once an ERC20 plugin code is deployed, it can be reused by any tokens that support the 1inch plugin standard.
+
+Support for plugins on the token side is similar to the implementation of classic ERC20 extensions (i.e., OpenZeppelin ERC20 extensions). The deployment and usage are permissionless from the perspective of a token contract owner, since the holder is the actor who decides which plugin to subscribe to.
```
[the remark this answers is not in the document]
Comment (tradersnow222) [reply README.md]: fixed

File: README.md [hunk 8]
```diff
@@ -1,69 +1,186 @@
...
+## Primary Benefits
+
+- **100% permissionless from the token contract owner**: Open to all participants.
+- **Risk-free participation**: Token plugins do not require any approval, deposit, or transfer of funds into an external contract for participation.
+- **Multiple plugin connections**: Users can connect with multiple plugins, allowing for simultaneous involvement in multiple incentive programs or governance systems, etc. (subject to a predefined limit, set at deployment).
+- **Simple to adopt**: Implementation is only 150 lines of code.
+- **High security**: Audited by top-notch firms such as OpenZeppelin, providing the highest standard of security.
+- **Built-in reentrancy protection**: [ReentrancyGuard.sol](https://github.com/1inch/token-plugins/blob/master/contracts/libs/ReentrancyGuard.sol) is integrated.
+- **Custom ERC20 representation**: A plugin can be represented by its own associated ERC20 (custom inheritance), enabling a participant to receive benefits for simply holding the token.
+- **Future support**: NFT (ERC721) and Multi-token (ERC1155) support are coming soon!
```
[the remark this answers is not in the document]
Comment (tradersnow222) [reply README.md]: fixed

File: README.md [hunk 9]
```diff
@@ -1,69 +1,186 @@
...
+The token plugins standard is designed to be secure and to prevent asset loss, gas, and DoS attacks on transfers.
+
+## Primary Benefits
+
+- **100% permissionless from the token contract owner**: Open to all participants.
+- **Risk-free participation**: Token plugins do not require any approval, deposit, or transfer of funds into an external contract for participation.
+- **Multiple plugin connections**: Users can connect with multiple plugins, allowing for simultaneous involvement in multiple incentive programs or governance systems, etc. (subject to a predefined limit, set at deployment).
+- **Simple to adopt**: Implementation is only 150 lines of code.
+- **High security**: Audited by top-notch firms such as OpenZeppelin, providing the highest standard of security.
+- **Built-in reentrancy protection**: [ReentrancyGuard.sol](https://github.com/1inch/token-plugins/blob/master/contracts/libs/ReentrancyGuard.sol) is integrated.
```
[the remark this answers is not in the document]
Comment (tradersnow222) [reply README.md]: fixed

File: README.md [hunk 10]
```diff
@@ -1,69 +1,186 @@
...
+- **Custom ERC20 representation**: A plugin can be represented by its own associated ERC20 (custom inheritance), enabling a participant to receive benefits for simply holding the token.
+- **Future support**: NFT (ERC721) and Multi-token (ERC1155) support are coming soon!
+
+## Use-Cases
+
+Here are some examples of how Token Plugins is currently being (or could be used) today:
+
+### st1INCH resolver delegation
+Staked 1INCH tokens earn rewards from Resolvers in the Intent Swap system. In order to earn rewards, the tokens must be pointed to a specific Resolver. The resolver is incentivized to have st1INCH pointed at them, so they will reward delegators with some amount of funds. The delegation of st1INCH is done with a token plugin, so there is no need to transfer the tokens to another contract.
```
[the remark this answers is not in the document]
Comment (tradersnow222) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (tradersnow222) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (tradersnow222) [reply README.md]: fixed

File: README.md (hunk 4 above)
Comment (tradersnow222) [reply README.md]: fixed

File: README.md (hunk 5 above)
Comment (tradersnow222) [reply README.md]: fixed

File: README.md (hunk 6 above)
Comment (tradersnow222) [reply README.md]: fixed

File: README.md (hunk 1 above)
Comment (tradersnow222) [reply README.md]: fixed

File: README.md [hunk 11]
```diff
@@ -1,73 +1,108 @@
...
-2. Wallets can plugin: `MyToken.addPlugin(plugin)`, where `plugin` is address of your and third-party deployed plugin.
-3. Now every time wallet balance is change the plugin will know about it
+This will add support for the plugin infrastructure.
 
-## How to create your own plugin
+- **Wallets can plugin**: `MyToken.addPlugin(plugin)`, where `plugin` is the address of your or a third-party deployed plugin.
+  - Now every time a wallet balance changes, the plugin will know about it.
 
-To create your own plugin it is necessary to inherit `Plugin` contract and implement its abstract function `_updateBalances`.
+### How do accounts (users) add or remove a plugin?
```
Comment (galekseev) [inline README.md:90]: Why do we need this section?

**Review by galekseev — CHANGES_REQUESTED** [review]

File: README.md (hunk 11 above)
Comment (tradersnow222) [reply README.md]: Section removed

File: README.md [hunk 12]
```diff
@@ -1,73 +1,108 @@
...
 
-- [FarmingPlugin](https://github.com/1inch/farming) - *The 1inch staking reward farming is based on this plugin.*
-This plugin introduces farming without the need for asset transfer or locking on a farm. An account owner can join the farm by simply adding this plugin to their account. The plugin handles all necessary accounting for joined farmers and distributes rewards based on the actual farming token balance of the account during the farming period. This is particularly useful for non-transferable tokens, for example such as debt tokens in lending protocols.
-- [DelegatingPlugin](https://github.com/1inch/delegating) - *The 1inch fusion mode resolver delegation is based on this plugin.*
-This plugin allows an account balance to be delegated to another address. This can be beneficial for governance tokens if an account owner wants to delegate their voting power to another account without a physical token transfer. The owner can recall or redelegate their delegation at any time without locking their tokens in a governance contract.
+## Implementation
 
-## Security and limitations
+![ERC20Plugins](/src/img/PluginsDiagram.png)
```
[the remark this answers is not in the document]
Comment (tradersnow222) [reply README.md:67]: Fixed.

**Review by galekseev — APPROVED** [review]

**Pull request merged**